src/main/java/com/example/javaitmo2/service/CarService.java

```diff
@@ -23,7 +29,24 @@ public CarService(CarRepository repository, ModelMapper modelMapper) {
     }
 
     public ArrayList<CarResponse> getList() {
-        List<CarEntity> list = repository.findAll();
+        Iterable<CarEntity> list = repository.findAll();
+        ArrayList<CarResponse> result = new ArrayList<>();
+        for (CarEntity car: list) {
+            result.add(modelMapper.map(car, CarResponse.class));
+        }
+        return result;
+    }
+
+    public ArrayList<CarResponse> search(String brand, Integer minSeatsCount, Integer page, Integer size) throws ValidationException {
+        Collection<CarEntity> list;
+        if (brand != null) {
+            list = repository.findAllByBrandOrderByIdDesc(brand);
+        } else if (minSeatsCount != null) {
+            Pageable pager = PageRequest.of(page, size, Sort.by("id").descending());
+            list = repository.findAllBySeatsCountGreaterThan(minSeatsCount, pager);
+        } else {
+            throw new ValidationException("No parameters founded");
+        }
         ArrayList<CarResponse> result = new ArrayList<>();
         for (CarEntity car: list) {
             result.add(modelMapper.map(car, CarResponse.class));
```

Comment:
Try to use functional approach oftenly.
list.stream().map...

Reply:
+